### PR TITLE
[Integrations] Support PyTorch OpenEnv

### PR DESCRIPTION
**Pull request by lynnliu030**
Integrate [Pytorch OpenEnv](https://github.com/meta-pytorch/OpenEnv) with SkyRL, register it as a SkyRL-Gym environment and use SkyRLGymGenerator. 

* Support environment such as coding, echo, finrl, openspiel, atari, and sumo RL. 
* Provide example training scripts (with dummy train / validation data) for coding and echo environment 

Main integration is in `integrations/openenv/env.py` 

Example reward:
image

### CODE REVIEW

File: skyrl-train/pyproject.toml [hunk 1]
```diff
@@ -101,7 +101,7 @@ docs = [
 ]
 # TODO(tgriggs): Add `sandboxes` here once available as a package.
 sandboxes = [
-    "litellm[proxy]>=1.67.5",
+    "litellm>=1.75.5",
```
[the remark this answers is not in the document]
Comment (lynnliu030) [reply skyrl-train/pyproject.toml]: `litellm` version incompatible with https://github.com/meta-pytorch/OpenEnv/blob/main/pyproject.toml; any better way to deal with this?

File: skyrl-train/docs/examples/pytorch_openenv.rst [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    uv run examples/openenv/dummy_openenv_dataset.py --output_dir ~/data/openenv --env_name echo_env
+    # Or generate datasets for all environments:
+    # uv run examples/openenv/dummy_openenv_dataset.py --output_dir ~/data/openenv 
+
+This creates training and validation datasets with example prompts for the specified environment. Available environments include: ``echo_env``, ``coding_env``, ``openspiel-env``, ``atari-env``, ``sumo-rl-env``, ``finrl-env``.
+
+Training
+--------
+
+We provide an example training script for Qwen2.5-0.5B-Instruct on OpenEnv environments:
```
Comment (SumanthRH) [inline skyrl-train/docs/examples/pytorch_openenv.rst]:
```suggestion
We provide an example training script for Qwen2.5-1.5B-Instruct on OpenEnv environments:
```

File: skyrl-train/docs/examples/pytorch_openenv.rst [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+.. code-block:: bash
+    cd skyrl-train
+    # Execute from skyrl-train directory
+    bash examples/openenv/run_dummy_openenv.sh
```
Comment (SumanthRH) [inline skyrl-train/docs/examples/pytorch_openenv.rst]:
```suggestion
    bash examples/openenv/run_openenv.sh
```

File: skyrl-train/docs/examples/pytorch_openenv.rst [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+You can customize the training by setting environment variables:
+
+.. code-block:: bash
+
+    ENV_NAME=coding_env NUM_GPUS=2 bash examples/openenv/run_dummy_openenv.sh
```
Comment (SumanthRH) [inline skyrl-train/docs/examples/pytorch_openenv.rst]:
```suggestion
    ENV_NAME=coding_env NUM_GPUS=2 bash examples/openenv/run_openenv.sh
```

File: skyrl-train/docs/examples/pytorch_openenv.rst [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+
+Supporting environments are: ``echo_env``, ``coding_env``, ``openspiel-env``, ``atari-env``, ``sumo-rl-env``, ``finrl-env``.
+
+Tips
+~~~~~
+
+- **Docker Resources**: Ensure sufficient Docker resources are available, especially for computationally intensive environments like Atari or OpenSpiel.
+- **Generation Format**: The generation format is expected to be a single action wrapped in ``<action>...</action>`` tags for dummy testing. Change `_get_openenv_action` in :code_link:`skyrl_gym/envs/openenv/env.py` for this parsing logic.
```
Comment (SumanthRH) [inline skyrl-train/docs/examples/pytorch_openenv.rst]:
```suggestion
- **Generation Format**: The generation format is expected to be a single action wrapped in ``<action>...</action>`` tags for dummy testing. Change `_get_openenv_action` in :code_link:`skyrl_train/integrations/openenv/env.py` for this parsing logic.
```

File: skyrl-train/docs/examples/pytorch_openenv.rst [hunk 6]
```diff
@@ -0,0 +1,104 @@
+SkyRL + PyTorch OpenEnv: Training a RL Agent in OpenEnv
+===========================================================
+
+In this example, we walk through a simple example on how to train a reinforcement learning agent using SkyRL with `PyTorch OpenEnv <https://github.com/meta-pytorch/OpenEnv>` environments. OpenEnv provides isolated execution environments for agentic RL training with Gymnasium-style APIs.
+
+How does it work?
+------------------
+
+SkyRL integrates with OpenEnv through a custom environment wrapper ``OpenEnv`` that implements the ``BaseTextEnv`` interface. This wrapper allows SkyRL to interact with various OpenEnv environments including:
```
Comment (SumanthRH) [inline skyrl-train/docs/examples/pytorch_openenv.rst]:
```suggestion
SkyRL integrates with any Gymnasium API-based environment easily with SkyRL-Gym, which provides a simple interface for text-based environments called ``BaseTextEnv``. We integrate OpenEnv environments through a custom environment wrapper ``OpenEnv`` that implements the ``BaseTextEnv`` interface. This wrapper allows SkyRL to interact with various OpenEnv environments, including:
```

File: skyrl-train/integrations/openenv/prepare_dummy_dataset.py [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+    # Shared instruction appended to each example
+    instruction = "Wrap the action between <action> and </action> tags.\n" "For example: <action>ACTION_HERE</action>."
+
+    if env_name == "echo_env":
+        # Echo environment simply echoes the text back.
+        examples = [
+            "Say hello to the environment.",
+            "Send a short test message.",
+            "Write one more message to echo back.",
```
Comment (SumanthRH) [inline skyrl-train/integrations/openenv/prepare_dummy_dataset.py]: Why do these instructions need to be different? I thought hte env will generate a random message to echo back or something?

Comment (lynnliu030) [reply skyrl-train/integrations/openenv/prepare_dummy_dataset.py]: Fixed

File: skyrl-train/integrations/openenv/README.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+For issues with SkyRL or the integration with OpenEnv, please [open an Issue](https://github.com/NovaSky-AI/SkyRL/issues/new). 
+
+### Datasets
+
+We use dummy datasets for all the environment integration now. Please modify `prepare_dummy_dataset.py` as needed to extract and prepare the correct datasets.
+
+## TODOs and Limitations
+We welcome any contributions to help resolve the remaining tasks!
+* Make it easier to specify different OpenEnv environments used for training and validation.
+* Make it smoother to specify which dataset splits to use (ie, resolve the challenge specified in the `Troubleshooting` section.)
```
Comment (SumanthRH) [inline skyrl-train/integrations/openenv/README.md]:
```suggestion
* Make it smoother to specify which dataset splits to use
```

File: skyrl-train/docs/examples/openenv.rst [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+.. code-block:: bash
+
+    ENV_NAME=echo_env NUM_GPUS=2 bash integrations/openenv/run_openenv.sh
+
+
+Supporting environments are: ``echo_env``, ``coding_env``, ``openspiel-env``, ``atari-env``, ``sumo-rl-env``, ``finrl-env``.
+
+Example Reward Curve 
+--------
+Example reward curve for the ``echo_env`` environment looks like this:
```
Comment (SumanthRH) [inline skyrl-train/docs/examples/openenv.rst]:
```suggestion
Here's how the reward curve for the above example script looks like after a few steps: 
```

**Review by SumanthRH — APPROVED** [review]
LFG
